### PR TITLE
Feature Authentication Develop

### PR DESCRIPTION
**Pull request by hirbabar**
1. Login Flow -  Fe and BE
2. Switch flow - FE and BE

### CODE REVIEW

File: customapi/src/test/java/com/publicissapient/kpidashboard/apis/auth/token/TokenAuthenticationServiceImplTest.java [hunk 1]
```diff
@@ -204,11 +204,11 @@ public void invalidateAuthToken() {
 		verify(userTokenReopository, times(1)).deleteByUserNameIn(users);
 	}
 
-	@Test
+	/*@Test
 	public void setUpdateAuthFlagForExpDateNull() {
```
Comment (Chittauri) [inline customapi/src/test/java/com/publicissapient/kpidashboard/apis/auth/token/TokenAuthenticationServiceImplTest.java:208]: either remove the code or remove @test

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/auth/model/ActionPoliciesDTO.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package com.publicissapient.kpidashboard.apis.auth.model;
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/auth/model/ActionPoliciesDTO.java:1]: add copy right

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/auth/model/ActionPoliciesDTO.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import com.publicissapient.kpidashboard.common.model.rbac.UserInfoDTO;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+/**
+ * Class to hold request data for save action policy.
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/auth/model/ActionPoliciesDTO.java:14]: provide proper description if this class is not required in whole code, do not create it. if required then why is this required provide proper description

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/auth/rest/ForgotPasswordController.java [hunk 4]
```diff
@@ -14,7 +14,8 @@
  * See the License for the specific language governing permissions and
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/auth/rest/ForgotPasswordController.java:14]: if not required, then remove the class or if required then provide proper description so that no one in future should delete this

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/auth/token/TokenAuthenticationServiceImpl.java [hunk 5]
```diff
@@ -95,26 +98,49 @@ public class TokenAuthenticationServiceImpl implements TokenAuthenticationServic
...
+
+	public String createJwtToken(Authentication authentication) {
+		return Jwts.builder().setSubject(authentication.getName())
+				.claim(DETAILS_CLAIM, authentication.getDetails())
+				.claim(ROLES_CLAIM, getRoles(authentication.getAuthorities()))
+				.setExpiration(new Date(System.currentTimeMillis() + tokenAuthProperties.getExpirationTime()))
+				.signWith(SignatureAlgorithm.HS512, tokenAuthProperties.getSecret()).compact();
 	}
 
 	@SuppressWarnings("unchecked")
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/auth/token/TokenAuthenticationServiceImpl.java:121]: provide proper description around the function

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/UserTokenAuthenticationDTO.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import lombok.Builder;
+import lombok.Data;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.util.List;
+
+/**
+ * aksshriv1
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/UserTokenAuthenticationDTO.java:30]: proper description of the class is missing

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/CacheService.java [hunk 7]
```diff
@@ -101,4 +101,9 @@ public Object getFromApplicationCache(String[] keyList, String kpiSource, Intege
 
 	Map<String, AdditionalFilterCategory> getAdditionalFilterHierarchyLevel();
 
+	//-- auth-N-auth changes starts here---
+	//List<ActionPoliciesDTO> getActionPoliciesFromCache();
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/CacheService.java:105]: ActionPoliciesDTO is not used then y required in the code

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/impl/CacheServiceImpl.java [hunk 8]
```diff
@@ -257,4 +260,16 @@ public Map<String, AdditionalFilterCategory> getAdditionalFilterHierarchyLevel()
...
\ No newline at end of file
+
+}
+	/**
+	 *
+	 * @param
+	 * @return
+	 */
+	/*@Override
+	@Cacheable(value = CommonConstant.ACTION_POLICY_RULES_CACHE)
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/common/service/impl/CacheServiceImpl.java:271]: if not required, then remove the code, or else provide the proper description so that no one in future revert the code

File: customapi/src/main/java/com/publicissapient/kpidashboard/apis/model/ServiceResponse.java [hunk 9]
```diff
@@ -25,6 +25,11 @@ public class ServiceResponse extends BaseResponse {
 
 	private Object data;
 
+	// --- auth-N-auth required code starts here --------------
+	public ServiceResponse() {
```
Comment (Chittauri) [inline customapi/src/main/java/com/publicissapient/kpidashboard/apis/model/ServiceResponse.java:29]: y this constructor required???

**Review by risshukl0 — APPROVED** [review]

**Review by brahmanand1 — APPROVED** [review]
